modules/core/src/test/java/org/apache/ignite/internal/processors/service/ServicePredicateAccessCacheTest.java

```diff
@@ -75,36 +76,39 @@
     public void testPredicateAccessCache() throws Exception {
         final Ignite ignite0 = startGrid(0);
 
-        ignite0.getOrCreateCache(new CacheConfiguration<>(DEFAULT_CACHE_NAME)
-            .setName("testCache")
-            .setAtomicityMode(ATOMIC)
-            .setCacheMode(REPLICATED)
-            .setWriteSynchronizationMode(FULL_SYNC));
+        CacheConfiguration<String, String> cacheCfg  = new CacheConfiguration<>();
+
+        cacheCfg.setName("testCache");
+        cacheCfg.setAtomicityMode(ATOMIC);
+        cacheCfg.setCacheMode(REPLICATED);
+        cacheCfg.setWriteSynchronizationMode(FULL_SYNC);
+
+        IgniteCache<String, String> cache = ignite0.getOrCreateCache(cacheCfg);
+
+        cache.put(ignite0.cluster().localNode().id().toString(), "val");
 
         latch = new CountDownLatch(1);
 
-        final ClusterGroup grp = ignite0.cluster().forPredicate(new IgnitePredicate<ClusterNode>() {
-            @Override public boolean apply(ClusterNode node) {
-                System.out.println("Predicated started [thread=" + Thread.currentThread().getName() + ']');
+        final ClusterGroup grp = ignite0.cluster().forPredicate((IgnitePredicate<ClusterNode>) node -> {
```

Comment:
Why this change?

Reply:
The reason is serialization. This predicate will be used in ServiceConfiguration and send over a network.